src/types/model.ts

```diff
@@ -113,6 +113,69 @@ export interface Dataset {
   alternativeDataSharingPlanFile: FileStorageObject;
 }
 
+interface DacTerm {
+  dacId: number;
+  dacName: string;
+  dacEmail: string;
+}
+
+interface InstitutionTerm {
+  id: number;
+  name: string;
+}
+
+interface UserTerm {
+  userId: number;
+  displayName: string;
+  institution: InstitutionTerm;
+}
+
+export interface StudyTerm {
+  description: string;
+  studyName: string;
+  studyId: number;
+  phsId: string;
+  phenotype: string;
+  species: string;
+  piName: string;
+  dataSubmitterEmail: string;
+  dataSubmitterId: number;
+  dataCustodianEmail: string[];
+  publicVisibility: boolean;
+  dataTypes: string[];
+}
+
+interface DataUseTerm {
+  code: string;
+  description: string;
+}
+
+interface DataUseSummary {
+  primary: DataUseTerm[];
+  secondary: DataUseTerm[];
+}
+
+export interface DatasetTerm {
+  datasetId: number;
+  createUserId: number;
+  createUserDisplayName: string;
+  datasetIdentifier: string;
+  deletable: boolean;
+  datasetName: string;
+  participantCount: number;
+  dataUse: DataUseSummary;
+  dataLocation: string;
+  url: string;
+  dacId: number;
+  dacApproval: boolean;
+  accessManagement: string;
+  approvedUserIds: number[];
+  study: StudyTerm;
+  submitter: UserTerm;
+  updateUser: UserTerm;
+  dac: DacTerm;
+}
+
```

Comment:
I'd love to hear more about the design of your approach. I'm wondering about the distinctions between this file and the constants file. Additionally, I'm wondering about the abstraction to the Display file.

Reply:
overall, this looks great and the video looks good, just looking for a bit more info for my understanding!

Reply:
@rjohanek great question! The goal was to break up and separate concerns. Previously, the `DatasetSearchTable` handled a lot - I wanted to reduce what it handled. The first abstraction was to pull out the table. Eventually I thought pulling out the constants would be nice as well, so `DatasetSearchTableDisplay` just handled the rendering of the table, and `DatasetSearchTableConstants` handled the constant values describing the rendering.

`model.ts` is specifically the type definitions of the API responses, and doesn't include any scope beyond that.